frontend/src/workspace/funbar/PopupCanvas.js

```diff
@@ -0,0 +1,217 @@
+        <PortalFunction {...props} />
+        <Buttons {...props}/>
+    </Modal>
+  );
+
+  function Background(props) {
+    return (
+      <div
+        style={{
+          position: "absolute",
+          alignSelf: 'center',
+          top: props.top + popupDimensions.canvasY,
+          width: popupDimensions.canvasWidth,
+          height: popupDimensions.canvasHeight,
+          backgroundColor: "white",
+          opacity: "0.85",
+          borderRadius: 30,
+        }}
+      />
+    );
+  }
+
+  function PortalTextBox(props) {
+    return (
+      <div // Text box: "Enter Name of Image"
+        style={{
+          position: "absolute",
+          alignSelf: 'center',
+          top: props.top + popupDimensions.textfieldY,
+          fontSize: gui.popTextFontSize,
+          fontFamily: gui.functionFont,
+          width: popupDimensions.textfieldWidth,
+          height: popupDimensions.textfieldHeight,
+          textAlign: "center",
+          alignItems: "center",
+        }}
+      >
+        <input
+          type={"text"}
+          placeholder={"Enter Name Of Image"}
+          style={{
+            width: popupDimensions.textfieldWidth,
+            height: popupDimensions.textfieldHeight,
+            border: "2px solid #008CBA",
+            textAlign: "center",
+          }}
+          onChange={(e) => props.setImageName(e.target.value)}
+        />
+      </div>
+    );
+  }
+
+  function PortalImage(props) {
+    return (
+      <div style={{
+        position: 'absolute',
+        alignSelf: 'center',
+        alignItems: 'center',
+        top: props.top + popupDimensions.imageY,
+      }}>
+        <MISTImage
+          x={-popupDimensions.imageWidth/2}
+          y={0}
+          width={popupDimensions.imageWidth}
+          height={popupDimensions.imageHeight}
+          renderFunction={props.renderFunction.renderFunction}
+          automated={true}
+        />
+      </div>
+    );
+  }
+
+  function PortalFunction(props) {
+    return (
+      <div
+        style={{
+          position: "absolute",
+          alignSelf: 'center',
+          top: props.top + popupDimensions.rfTextY,
+          fontSize: 20,
+          textAlign: "center",
+          verticalAlign: "middle",
+          width: popupDimensions.rfTextWidth,
+          height: popupDimensions.rfTextHeight,
+          overflow: "auto",
+          fill: "#FFF",
+          stroke: "#000",
+        }}
+      >
+        <p
+          style={{
+            width: popupDimensions.rfTextWidth,
+            height: popupDimensions.rfTextHeight,
+            verticalAlign: "middle",
+          }}
+        >
+          {props.renderFunction.renderFunction}
+        </p>
+      </div>
+    );
+  }
+
+  function Buttons(props) {
+    const [imageExists, setImageExists] = useState("initial");
+
+    function SaveImage() {
+      //var newName = props.imageName;
+      //newName = removeOuterWhiteSpace(newName);
+      //var response = getImageExists(newName);
```

Comment:
These don't seem necessary anymore. If that is the case, you should remove them.